modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridDhtAtomicCache.java

```diff
@@ -2734,9 +2735,22 @@ else if (GridDhtCacheEntry.ReaderId.contains(readers, nearNode.id())) {
                     continue;
                 }
 
-                if (storeErr != null &&
-                    storeErr.failedKeys().contains(entry.key().value(ctx.cacheObjectContext(), false)))
-                    continue;
+                if (storeErr != null) {
+                    Object key = entry.key();
+
+                    try {
+                        key = entry.key().value(ctx.cacheObjectContext(), false);
+                    }
+                    catch (BinaryInvalidTypeException e) {
+                        if (log.isDebugEnabled()) {
+                            log.warning("Failed to get a value of the key: " + key +
```

Comment:
Without log throttle it is a real possibility to generate gigabytes of logs per minute.

I don't think that this is sound approach.